src/components/CIPipelineN/CIPipeline.tsx

```diff
@@ -0,0 +1,695 @@
+import React, { useState, useEffect, createContext } from 'react'
+import { NavLink } from 'react-router-dom'
+import { ButtonWithLoader, ConditionalWrap, DeleteDialog, showError, VisibleModal } from '../common'
+import { Redirect, Route, Switch, useParams, useRouteMatch, useLocation, useHistory } from 'react-router'
+import { BuildStageVariable, BuildTabText, TriggerType, ViewType } from '../../config'
+import {
+    deleteCIPipeline,
+    getGlobalVariable,
+    getInitData,
+    getInitDataWithCIPipeline,
+    saveCIPipeline,
+} from '../ciPipeline/ciPipeline.service'
+import { toast } from 'react-toastify'
+import { ServerErrors } from '../../modals/commonTypes'
+import { ValidationRules } from '../ciPipeline/validationRules'
+import {
+    CIPipelineDataType,
+    FormType,
+    PluginType,
+    RefVariableStageType,
+    RefVariableType,
+    ScriptType,
+    StepType,
+    TaskErrorObj,
+    VariableType,
+} from '../ciPipeline/types'
+import { ReactComponent as Close } from '../../assets/icons/ic-close.svg'
+import Tippy from '@tippyjs/react'
+import { PreBuild } from './PreBuild'
+import { Sidebar } from './Sidebar'
+import { Build } from './Build'
+import { ReactComponent as AlertTriangle } from '../../assets/icons/ic-alert-triangle.svg'
+
+interface CIPipelineType {
+    appName: string
+    connectCDPipelines: number
+    getWorkflows: () => void
+    close: () => void
+}
+
+export const ciPipelineContext = createContext(null)
+
+export default function CIPipeline({ appName, connectCDPipelines, getWorkflows, close }: CIPipelineType) {
+    let { appId, workflowId, ciPipelineId } = useParams<{ appId: string; workflowId: string; ciPipelineId: string }>()
```

Comment:
use `const`

Reply:
okay I see that ciPipelineId is being modified

Reply:
ciPipelineId is getting reassigned on next line